test/Table_Tests/src/In_Memory/Split_Tokenize_Spec.enso

```diff
@@ -248,10 +248,34 @@ spec =
             actual = t.parse_to_columns "bar" "(\d)(\d)" parse_values=False
             actual.should_equal_verbose expected
 
-        Test.specify "no matches" <|
-            t = Table.from_rows ["foo", "bar", "baz"] [["x", "12 34p q56", "y"], ["xx", "a48 59b", "yy"]]
-            expected = Table.from_rows ["foo", "baz"] []
-            actual = t.parse_to_columns "bar" "asdf"
+        Test.specify "column name clash" <|
+            t = Table.from_rows ["foo", "bar", "bar 1"] [["x", "12 34p q56", "y"], ["xx", "a48 59b", "yy"]]
+            expected = Table.from_rows ["foo", "bar 0", "bar 1_1", "bar 1"] [["x", 1, 2, "y"], ["x", 3, 4, "y"], ["x", 5, 6, "y"], ["xx", 4, 8, "yy"], ["xx", 5, 9, "yy"]]
+            actual = t.parse_to_columns "bar" "(\d)(\d)"
+            actual.should_equal_verbose expected
+
+        Test.specify "column and group name clash" <|
+            t = Table.from_rows ["foo", "bar", "baz"] [["x", "123", "y"]]
+            expected = Table.from_rows ["foo", "bar", "baz_1", "quux", "baz"] [["x", 1, 2, 3, "y"]]
+            actual = t.parse_to_columns "bar" "(?<bar>\d)(?<baz>\d)(?<quux>\d)"
+            actual.should_equal_verbose expected
+
+        Test.specify "input with no matches" <|
+            t = Table.from_rows ["foo", "bar", "baz"] [["x", "a", "y"]]
+            expected = Table.from_rows ["foo", "bar", "baz"] []
+            actual = t.parse_to_columns "bar" "\d+"
+            actual.should_equal_verbose expected
```

Comment:
I'd also appreciate a case where the _first_ row has no matches but the second one does match something.

I know it's a bit convoluted, but I saw some logic relying on the result of the _first_ row somewhere in your code (not sure if that didn't change in the meantime) - and so IMO it would be good to ensure that it does not somehow rely on the first input not matching. So I think worth adding a test with 2+ rows where first row is non-empty but has no matches and possibly also where the first row is `Nothing`.

Reply:
That logic is now gone; previously I was inspecting the first row to derive the columns, and now the caller must specify the columns explicitly.

Reply:
Okay